File: IT/Containers/Esercizi/C++/esercizio4.cpp
/*
file esercizio4.cpp
compilare con: g++ -std=c++11 esercizio4.cpp
*/

#include <iostream>
#include <vector>
#include <iterator>

using std::cout;
using std::endl;
using std::vector;
using std::ostream_iterator;

int main()
{
	vector<int> nums = {1, 3, 5, 7, 9};

	cout << "[";
	copy(nums.begin(), nums.end(), ostream_iterator<int>(cout, ", "));
	cout << "\b\b]" << endl;

	int tot = 0;

	for (int n : nums) {
		tot += n;
	}

	cout << "Somma degli elementi: " << tot << endl;
}

File: IT/1/Containers/Esercizi/Java/Esercizio4.java
/* file Esercizio4.java */

import java.util.ArrayList;

public class Esercizio4 {
	public static void main(String [] args) {
		ArrayList<Integer> nums = new ArrayList<Integer>();

		nums.add(1);
		nums.add(3);
		nums.add(5);
		nums.add(7);
		nums.add(9);

		System.out.println(nums);

		int tot = 0;

		for (int n : nums) {
			tot += n;
		}

		System.out.println("Somma degli elementi: " + tot);
	}
}

